internal/newusers/manager.go

```diff
@@ -0,0 +1,338 @@
+// Package newusers is a transitional package that will replaces the users one.
+package newusers
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"os/user"
+	"strconv"
+	"syscall"
+	"time"
+	_ "unsafe" //nolint:revive,nolintlint // needed for go:linkname. nolintlint as false positive then.
+
+	"github.com/ubuntu/authd/internal/newusers/cache"
+	"github.com/ubuntu/authd/internal/users"
+	"github.com/ubuntu/decorate"
+)
+
+const (
+	// defaultEntryExpiration is the amount of time the user is allowed on the cache without authenticating.
+	// It's equivalent to 6 months.
+	defaultEntryExpiration = time.Hour * 24 * 30 * 6
+
+	// defaultCleanupInterval is the interval upon which the cache will be cleaned of expired users.
+	defaultCleanupInterval = time.Hour * 24
+)
+
+// dirtyFlagDbName is the dirty flag exported for tests.
+//
+//go:linkname dirtyFlagDbName github.com/ubuntu/authd/internal/newusers/cache.dirtyFlagDbName
+var dirtyFlagDbName string
```

Comment:
No, we should not use that in production :) For tests, this is fine, but let’s not abuse Go itself for production code!

Reply:
I agree... I wasn't happy with it either, but I also wasn't too happy about making it public. Do you have any ideas?